dali/operators/reader/loader/nemo_asr_loader.cc

```diff
@@ -94,10 +94,16 @@ void NemoAsrLoader::PrepareMetadataImpl() {
 }
 
 void NemoAsrLoader::Reset(bool wrap_to_shard) {
-  current_index_ = wrap_to_shard ? start_index(shard_id_, num_shards_, SizeImpl()) : 0;
+  current_index_ = wrap_to_shard ? start_index(virtual_shard_id_, num_shards_, SizeImpl()) : 0;
   current_epoch_++;
 
   if (shuffle_after_epoch_) {
+    if (IsCheckpointingEnabled()) {
+      // With checkpointing enabled dataset order must be easy to restore.
+      // Shuffling is run with different seed every epoch, so this doesn't
+      // reduce the randomness.
```

Comment:
We probably should add a small note to release notes that enabling checkpointing for this reader changes the order or something like this as this might be important for the end user.